src/v3/src/util/deviceFingerprintingUtils.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2023-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+const getUserAgent = (): string => navigator.userAgent;
+
+const isWindowsPhone = (userAgent: string): RegExpMatchArray | null => userAgent.match(/windows phone|iemobile|wpdesktop/i);
+
+export const isMessageFromCorrectSource = (iframe: HTMLIFrameElement, event: MessageEvent)
+: boolean => (
+  event.source === iframe.contentWindow
+);
+
+// NOTE: This utility is similar to the DeviceFingerprinting.js file used for V2 authentication flows.
+export const generateDeviceFingerprint = (oktaDomainUrl: string): Promise<string> => (
+  // eslint cannot detect that setTimeout() at end of function will eventually return with Error rejection
+  // eslint-disable-next-line consistent-return
+  new Promise((resolve, reject) => {
```

Comment:
Actually, if you just remove the `return` from all places that uses `resolve(...)` or `reject(...)`. This is what is causing the linting issue.

Reply:
Yeah so we can do that to bypass the linting issue, but I believe the gen2 code purposefully added `return` statements specifically in the `userAgent` checks at the very beginning of the function because without the explicit `return`, the control flow will proceed past these `reject` calls and still instantiate the `iframe` but never clean it up, which we don't want to happen. Here's the gen2 code:
https://github.com/okta/okta-signin-widget/blob/acd4c0dc62b24bc9ae83f48cdda36f39c57f14df/src/v2/view-builder/utils/DeviceFingerprinting.js#L32-L36

The gen2 code would fail this eslint rule because it also isn't consistent with returning, but I thought it would be safer to keep this behavior and immediately return control and not create the `iframe` if we know right away that the `userAgent` check failed, so I ended up adding `return` to all locations where we `resolve` and `reject`, but `eslint` doesn't know about the `setTimeout` fallback logic, which will always return after 2 seconds if no other `return` is hit.

Does this make sense?

Reply:
One alternative that I _think_ could work would be to move the `userAgent` and `windowsPhone` checks below the instantiation of the `iframe` and then use `handleError` (which depends on the `iframe` being instantiated since it calls `removeIFrame()`) instead of `return reject ...`, but then we're instantiating the `iframe` in cases where we could have detected that it should have never been instantiated in the first place

Reply:
Ah I see, could you place a single return statement after the reject/resolve callbacks? Not sure of that would appease the lint error.

Reply:
I was able to refactor by moving the `userAgent` and `isWindowsPhone` checks outside of the overall `return new Promise...` and now the lint error isn't throwing, so this should be resolved!